src/components/TooltipNew/GuideTooltip.js

```diff
@@ -0,0 +1,272 @@
+
+const baseClass = 'guide-tooltip';
+
+const cx = classNames.bind(styles);
+
+const addPadding = (rect, padding) => {
+  const x = Math.round(rect.x) - padding;
+  const y = Math.round(rect.y) - padding;
+  const width = Math.round(rect.width) + 2 * padding;
+  const height = Math.round(rect.height) + 2 * padding;
+  const top = y;
+  const left = x;
+  const bottom = top + height;
+  const right = left + width;
+
+  return {
+    x,
+    y,
+    width,
+    height,
+    top,
+    left,
+    bottom,
+    right
+  };
+};
+
+class VirtualReference {
+  constructor(element) {
+    this.element = element;
+  }
+
+  getBoundingClientRect() {
+    return addPadding(this.element.getBoundingClientRect(), 8);
+  }
+
+  get clientWidth() {
+    return this.getBoundingClientRect().width;
+  }
+
+  get clientHeight() {
+    return this.getBoundingClientRect().height;
+  }
+}
+
+const GapOverlay = ({ gap, isVisible, slide }) => {
+  const overlayLeft = {
+    top: `${gap.top}px`,
+    left: '0',
+    width: `${gap.left}px`,
+    height: `${gap.height}px`
+  };
+  const overlayRight = {
+    top: `${gap.top}px`,
+    left: `${gap.right}px`,
+    width: `calc(100% - ${gap.right}px)`,
+    height: `${gap.height}px`
+  };
+  const overlayTop = {
+    top: '0',
+    left: '0',
+    width: '100%',
+    height: `${gap.top}px`
+  };
+  const overlayBottom = {
+    top: `${gap.bottom}px`,
+    left: '0',
+    width: '100%',
+    height: `calc(100% - ${gap.bottom}px)`
+  };
+
+  return (
+    <React.Fragment>
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayLeft}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayTop}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayRight}
+      />
+      <div
+        className={cx({
+          [styles[`${baseClass}__overlay`]]: true,
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayBottom}
+      />
+    </React.Fragment>
+  );
+};
+
+GapOverlay.propTypes = {
+  gap: PropTypes.shape({
+    top: PropTypes.number,
+    left: PropTypes.number,
+    bottom: PropTypes.number,
+    right: PropTypes.number,
+    width: PropTypes.number,
+    height: PropTypes.number
+  }),
+  isVisible: PropTypes.bool,
+  slide: PropTypes.bool
+};
+
+class GuideTooltip extends React.PureComponent {
```

Comment:
Was the components name consulted with Justyna? Maybe it should be `UserGuideTooltip`?

Reply:
She really didn't much care about the naming :D She said the `GuideTooltip` is fine.

Reply:
whatever, we should care about component names because the wrong name can confuse developers and it's gonna be harder for them to use the lib

Reply:
no problem - `UserGuideTooltip` it is :)